test/ae_mdw_web/controllers/contract_controller_test.exs

```diff
@@ -978,16 +978,18 @@ defmodule AeMdwWeb.Controllers.ContractControllerTest do
          [
            read_node_tx_details: fn
              _state, {^txi1, -1} ->
-               {contract_create_tx1, tx_hash1, :contract_create_tx, block_hash}
+               {contract_create_tx1, :contract_create_tx, tx_hash1, :contract_create_tx,
+                block_hash}
 
              _state, {^txi2, 0} ->
-               {contract_create_tx2, tx_hash2, :contract_call_tx, block_hash}
+               {contract_create_tx2, :contract_create_tx, tx_hash2, :contract_call_tx, block_hash}
 
              _state, {^txi3, -1} ->
-               {contract_create_tx3, tx_hash3, :contract_create_tx, block_hash}
+               {contract_create_tx3, :contract_create_tx, tx_hash3, :contract_create_tx,
+                block_hash}
 
              _state, {^txi4, -1} ->
-               {ga_attach_tx, tx_hash4, :ga_attach_tx, block_hash}
+               {ga_attach_tx, :contract_create_tx, tx_hash4, :ga_attach_tx, block_hash}
```

Comment:
With `{txi, -1}` it handles only `ga_meta_tx` and `paying_for` inner types. This would be a fix only on the test. I think `ga_attach` should continue without inner type tx.

Reply:
Could you explain a little bit further? What would you want the behavior to be here?

Reply:
It's just about the consistency of the mocking with the behavior of `read_node_tx_details` that doesn't return inner type for `ga_attach_tx`:
https://github.com/aeternity/ae_mdw/blob/master/lib/ae_mdw/db/util.ex#L177-L184

`:ga_attach_tx` in place of `:contract_create_tx` would be a more realistic test.

Reply:
With these branch changes, it does return `contract_create_tx` https://github.com/aeternity/ae_mdw/blob/channel-updates/lib/ae_mdw/db/util.ex#L178

Reply:
Please notice the condition `tx_type in [:ga_meta_tx, :paying_for_tx]` (`tx_type = :ga_attach_tx` doesn't have inner tx)

Reply:
You're right, I was confusing meta with attach. Done now, thanks!